packages/marketplace/src/services/apps.ts

```diff
@@ -1,27 +1,256 @@
-import { fetcher } from '@reapit/elements'
-import { URLS, generateHeader } from '../constants/api'
+import {
+  CreateAppModel,
+  PagedResultAppSummaryModel_,
+  AppDetailModel,
+  PagedResultAppRevisionModel_,
+  CreateAppRevisionModel,
+  AppRevisionModel,
+  ApproveModel,
+  RejectRevisionModel,
+  AppClientSecretModel,
+} from '@reapit/foundations-ts-definitions'
+import { fetcher, setQueryParams } from '@reapit/elements'
+import { URLS } from './constants'
+import { generateHeader } from './utils'
+import { logger } from 'logger'
+import { FetchByIdCommonParams, FetchListCommonParams } from './types'
 
-export interface FetchAppDetailParams {
-  id: string
+export interface FetchAppsListParams extends FetchListCommonParams {
+  developerId?: string[]
   clientId?: string
+  externalAppId?: string[]
+  category?: string[]
+  desktopIntegrationTypeId?: string
+  appName?: string
+  developerName?: string
+  companyName?: string
+  isFeatured?: boolean
+  isDirectApi?: boolean
+  onlyInstalled?: boolean
+  registeredFrom?: string
+  registeredTo?: string
+}
+
+export interface FetchAppByIdParams extends FetchByIdCommonParams {
```

Comment:
So just curious why below is type but this one is interface
So you can declare type extends like this `export type FetchAppByIdParams  = FetchByIdCommonParams & { clientId?: string }`